python/cudf/cudf/utils/cudautils.py

```diff
@@ -79,23 +30,27 @@ def find_index_of_val(arr, val, mask=None, compare="eq"):
     mask : mask of the array
     compare: str ('gt', 'lt', or 'eq' (default))
     """
-    found = cuda.device_array(shape=(arr.shape), dtype="int32")
-    if found.size > 0:
-        with _CUDFNumbaConfig():
-            if compare == "gt":
-                gpu_mark_gt.forall(found.size)(arr, val, found, arr.size)
-            elif compare == "lt":
-                gpu_mark_lt.forall(found.size)(arr, val, found, arr.size)
-            else:
-                if arr.dtype in ("float32", "float64"):
-                    gpu_mark_found_float.forall(found.size)(
-                        arr, val, found, arr.size
-                    )
+    arr = cudf.core.column.as_column(arr, nan_as_null=False)
+    target = cudf.core.column.arange(0, arr.size, dtype="int32")
+    if arr.size > 0:
```

Comment:
It looks like `found` is never defined if this conditional is not met, which I suspect would yield a `NameError` in the return line.